### PR TITLE
Transparent encryption and decryption for all tables that contain secrets or tokens

### PR DESCRIPTION
**Pull request by unknwon**
Encrypts and decrypts secrets columns in our database transparently.

Notes:

- `event_logs.argument` is left out, see [discussions in Slack](https://sourcegraph.slack.com/archives/C1JH2BEHZ/p1600862015004200).
- `repo.metadata` is not encrypted because we normalized most of its value and decided not worth encrypting.

Co-authored-by: Dax McDonald <31839142+daxmc99@users.noreply.github.com>

### CODE REVIEW

**Review by asdine — APPROVED** [review]
This looks good to me but I feel like the boilerplate needed to encrypt a single field is still a bit too big.
We could leverage the fact that the encryption functions are available from everywhere to try to make it as transparent as possible. I have proposed a lead in a comment but not sure if that's enough

File: cmd/repo-updater/repos/store.go [hunk 1]
```diff
@@ -757,18 +752,54 @@ ORDER BY id ASC LIMIT %s
...
+		}
+	}
+	*srcs = rawSrcs
+	return nil
+}
+
+type externalServiceRepo struct {
+	ExternalServiceID int64  `json:"external_service_id"`
+	RepoID            int64  `json:"repo_id"`
+	CloneURL          string `json:"clone_url"`
```
[the remark this answers is not in the document]
Comment (unknwon) [reply cmd/repo-updater/repos/store.go]: Nice idea (again)! PTAL https://github.com/sourcegraph/sourcegraph/pull/14214/commits/48f92f8e3544677a6ffda30a023f7cbbbf68f6c6, which simplified the code a lot.

**Comment by unknwon** [comment]
Since no real encryption is in place yet, there will be no harm to data if any application code behaves unexpected. Merging as-is and happy to address any post-merge comments!

File: cmd/repo-updater/repos/store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (asdine) [reply cmd/repo-updater/repos/store.go]: That looks great! Thank you!

**Pull request merged**